File: VacancyAccountingSystem/VacancyAccountingSystem/Models/Company.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace VacancyAccountingSystem.Models
{
    public class Company : User
    {
        public Company()
        {
            OpenedVacancies = new List<Vacancy>();
        }

        public string Name { get; set; }

        public ICollection<Vacancy> OpenedVacancies { get; set; }

        public string AboutCompany { get; set; }

        public string Website { get; set; }
    }
}

File: VacancyAccountingSystem/VacancyAccountingSystem/ClientApp/src/app/models/company.ts
import {Login} from './login';
import {Image} from './image';

export class Company {
  constructor(Login: Login, Name: string, Image: Image, Website: string, AboutCompany: string) {
    this.login = Login;
    this.image = Image;
    this.name = Name;
    this.website = Website;
    this.aboutCompany = AboutCompany;
  }

  id: number;
  name: string;
  aboutCompany: string;
  website: string;
  image: Image;
  login: Login;
}
